src/commands/files_upload.rs

```diff
@@ -0,0 +1,197 @@
+                .value_name("DISTRIBUTION")
+                .help("Optional distribution identifier for this file."),
+        )
+        .arg(
+            Arg::new("wait")
+                .long("wait")
+                .help("Wait for the server to fully process uploaded files."),
+        )
+        .arg(
+            Arg::new("file-headers")
+                .long("file-header")
+                .short('H')
+                .value_name("KEY VALUE")
+                .multiple_occurrences(true)
+                .help("Store a header with this file."),
+        )
+        .arg(
+            Arg::new("url_prefix")
+                .short('u')
+                .long("url-prefix")
+                .value_name("PREFIX")
+                .help("The URL prefix to prepend to all filenames."),
+        )
+        .arg(
+            Arg::new("url_suffix")
+                .long("url-suffix")
+                .value_name("SUFFIX")
+                .help("The URL suffix to append to all filenames."),
+        )
+        .arg(
+            Arg::new("ignore")
+                .long("ignore")
+                .short('i')
+                .value_name("IGNORE")
+                .multiple_occurrences(true)
+                .help("Ignores all files and folders matching the given glob"),
+        )
+        .arg(
+            Arg::new("ignore_file")
+                .long("ignore-file")
+                .short('I')
+                .value_name("IGNORE_FILE")
+                .help(
+                    "Ignore all files and folders specified in the given \
+                    ignore file, e.g. .gitignore.",
+                ),
+        )
+        .arg(
+            Arg::new("extensions")
+                .long("ext")
+                .short('x')
+                .value_name("EXT")
+                .multiple_occurrences(true)
+                .help(
+                    "Set the file extensions that are considered for upload. \
+                    This overrides the default extensions. To add an extension, all default \
+                    extensions must be repeated. Specify once per extension.",
+                ),
+        )
+}
+
+pub fn execute(matches: &ArgMatches) -> Result<()> {
+    let config = Config::current();
+    let release = config.get_release_with_legacy_fallback(matches)?;
+    let org = config.get_org(matches)?;
+    let project = config.get_project(matches).ok();
+    let api = Api::current();
+
+    let dist = matches.value_of("dist");
+    let mut headers = vec![];
+    if let Some(header_list) = matches.values_of("file-headers") {
+        for header in header_list {
+            if !header.contains(':') {
+                bail!("Invalid header. Needs to be in key:value format");
+            }
+            let mut iter = header.splitn(2, ':');
+            let key = iter.next().unwrap();
+            let value = iter.next().unwrap();
+            headers.push((key.trim().to_string(), value.trim().to_string()));
```

Comment:
why push onto headers in loop vs. using `.map().collect()`?

Reply:
Mapping over vector of results is imo just annoying and harder to read than simple loop.